test/boost/tablets_test.cc

```diff
@@ -2615,6 +2615,73 @@ SEASTAR_THREAD_TEST_CASE(test_per_shard_goal_mixed_dc_rf) {
     }, tablet_cql_test_config()).get();
 }
 
+// This test verifies that per-table tablet count is adjusted
+// in reaction to changes of relevant config and schema options.
+SEASTAR_THREAD_TEST_CASE(test_tablet_option_and_config_changes) {
+    auto cfg = tablet_cql_test_config();
+    cfg.db_config->tablets_initial_scale_factor(10.0);
+
+    do_with_cql_env_thread([] (auto& e) {
+        topology_builder topo(e);
+        auto dc = topo.dc();
+
+        // 3 shards. default initial scale wants 30 (32) tablets.
+        // keyspace 'initial' wants 2 tablets.
+        topo.add_node(node_state::normal, 3);
+
+        auto ks_name1 = add_keyspace(e, {{dc, 1}}, 2);
+        e.execute_cql(fmt::format("CREATE TABLE {}.table1 (p1 text, r1 int, PRIMARY KEY (p1)) "
+                                  "WITH tablets = {{}}", ks_name1)).get();
```

Comment:
why is `WITH tablets = {{}}` needed?

Reply:
It's not, just copy pasta. I'll remove it.